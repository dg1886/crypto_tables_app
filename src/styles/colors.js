export const COLORS = {
  LIGHT_GREY: "#383838",
  ERROR: "#d9002e",
  INHERIT: "inherit",

  DARK_THEME: {
    TEXT: "#ffffff",
    BACKGROUND: "#080808",
    SECONDARY: "#0f0f0f",
    BACKGROUND_ITEMS: "#1c1c1c",
    ACTIVE: "#fffffd",
    SECONDARY_BACKGROUND: "#838383",
    PRIMARY_RED: "#e36d85",
    SECONDARY_RED: "#ae5467",
  },
  LIGHT_THEME: {
    TEXT: "#080808",
    BACKGROUND: "#fffffd",
    SECONDARY: "#a49df1",
    BACKGROUND_ITEMS: "#d7dffd",
    ACTIVE: "#fffffd",
    SECONDARY_BACKGROUND: "#838383",
    BUTTON_BACKGROUND: "#7957c0",
    LIGHT_RED: "#ae5467",
  },
};

export const GRAPH_COLORS = {
  DARK_THEME: {
    FIRST: "#29a5e9",
    SECOND: "#ffffff",
    THIRD: "#ae5467",
  },
  LIGHT_THEME: {
    FIRST: "#29a5e9",
    SECOND: "#080808",
    THIRD: "#ae5467",
  },
};

export const CHART_COLORS = {
  DARK_THEME: {
    FIRST: "#4682b4",
    SECOND: "#1f2422",
    THIRD: "#ae5467",
  },
  LIGHT_THEME: {
    FIRST: "#4682b4",
    SECOND: "#d7dffd",
    THIRD: "#ae5467",
  },
};
