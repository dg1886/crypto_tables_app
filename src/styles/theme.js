export const baseTheme = {
  fontFamily: {
    primary: "Gilroy",
  },
  fontSize: {
    extraLarge: "36px",
    large: "24px",
    normal: "16px",
    small: "14px",
  },
  letterSpacing: {
    default: "1px",
  },
  iconSize: {
    height: "30px",
    width: "30px",
  },
  inputSize: {
    height: "45px",
    width: "400px",
  },
  colors: {
    dark: "#080808",
    darkGrey: "#0f0f0f",
    white: "#ffffff",
    grey: "#1c1c1c",
    lightGrey: "#838383",
    pink: "#e36d85",
    blue: "#29a5e9",
    red: "#ae5467",
  },
  boxShadow: {
    primary: "rgba(0, 0, 0, 0.24) 0px 3px 8px",
    secondary: "rgba(50, 50, 93, 0.25) 0px 50px 100px -20px, rgba(0, 0, 0, 0.3) 0px 30px 60px -30px",
  },
};
