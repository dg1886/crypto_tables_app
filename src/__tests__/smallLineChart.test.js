import { render, screen } from "@testing-library/react";
import TestRenderer from "react-test-renderer";
import { RecoilRoot } from "recoil";

import SmallLineChart, { lineCharts } from "../components/MainPageComponents/lineChart";
import NotificationProvider from "../services/errorContext";
import InfoForGraphContextProvider from "../services/infoForGraphContext";
import ThemeContextProvider from "../services/themeContext";
import { CHART_COLORS } from "../styles/colors";

const data = [
  {
    close: 56974.12,
    date: "2021-11-01T00:00:00.000Z",
    high: 69000,
    low: 53308.93,
    open: 61356.57,
    volume: 72246.48605716,
  },
  {
    close: 46214.37,
    date: "2021-12-01T00:00:00.000Z",
    high: 59099.64,
    low: 41967.5,
    open: 57027.28,
    volume: 71859.97407647,
  },
  {
    close: 38491.92,
    date: "2022-01-01T00:00:00.000Z",
    high: 47989,
    low: 32950.72,
    open: 46230,
    volume: 71750.65620092,
  },
  {
    close: 43178.98,
    date: "2022-02-01T00:00:00.000Z",
    high: 45850,
    low: 34324.05,
    open: 38483.56,
    volume: 57954.06442157,
  },
];

const lineChartMockColors = [
  CHART_COLORS.DARK_THEME.FIRST,
  CHART_COLORS.DARK_THEME.SECOND,
  CHART_COLORS.DARK_THEME.THIRD,
];

describe("smallLineChart", () => {
  it("number of line charts in a block", () => {
    expect(lineCharts.length)
      .toBe(4);
  });

  it("line chart colors", () => {
    const firstPrice = data[0]?.close;
    const lastPrice = data[data.length - 1]?.close;
    const penultPrice = data[data.length - 2]?.close;

    const profitPercent = (
      ((firstPrice * 100) / lastPrice - 100).toFixed(2)
    );
    const penultPercent = (
      ((firstPrice * 100) / penultPrice - 100).toFixed(2)
    );

    if (profitPercent >= penultPercent) {
      // eslint-disable-next-line jest/no-conditional-expect
      expect(lineChartMockColors[0])
        .toBe(CHART_COLORS.DARK_THEME.FIRST);
    }
    expect(lineChartMockColors[2])
      .toBe(CHART_COLORS.DARK_THEME.THIRD);
  });

  it("render without data", () => {
    render(
      <RecoilRoot>
        <ThemeContextProvider>
          <NotificationProvider>
            <InfoForGraphContextProvider>
              <SmallLineChart />
            </InfoForGraphContextProvider>
          </NotificationProvider>
        </ThemeContextProvider>
      </RecoilRoot>,
    );
    if (!data.length) {
      screen.getByText("No Data")
        .toBeInTheDocument();
    }
  });

  it("renders correctly", () => {
    const tree = TestRenderer
      .create(
        <RecoilRoot>
          <ThemeContextProvider>
            <NotificationProvider>
              <InfoForGraphContextProvider>
                <SmallLineChart />
              </InfoForGraphContextProvider>
            </NotificationProvider>
          </ThemeContextProvider>
        </RecoilRoot>,
      );
    expect(tree)
      .toMatchSnapshot();
  });
});
