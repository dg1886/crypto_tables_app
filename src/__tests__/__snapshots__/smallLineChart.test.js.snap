// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smallLineChart renders correctly 1`] = `
Array [
  <div
    className="sc-hKMtZM Wtjam"
  />,
  <div
    className="sc-jSMfEi eUtByO"
    color="#1f2422"
  >
    <div
      className="sc-gKXOVf bBKeoX"
    >
      <div
        className="sc-eCYdqJ sc-papXJ kJXsff dSWLEt"
      >
        <span
          className="sc-bczRLJ YxcWK"
        >
          BTC
        </span>
        <svg
          height="1.5rem"
          viewBox="0 0 24 24"
          width="1.5rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs>
            <style>
              .up { fill: #ae5467;} .down { fill: #29a5e9;}
            </style>
          </defs>
          <g
            id="a"
          />
          <g
            id="b"
          >
            <polygon
              className="up"
              points="8 14 16 14 12 19 8 14"
            />
            <polygon
              className="down"
              points="16 10 8 10 12 5 16 10"
            />
          </g>
        </svg>
        <span
          className="sc-bczRLJ YxcWK"
        >
          USD
        </span>
      </div>
      <div>
        <span
          className="sc-bczRLJ bYxRpA"
        >
          No Data
        </span>
      </div>
    </div>
  </div>,
  <div
    className="sc-jSMfEi eUtByO"
    color="#1f2422"
  >
    <div
      className="sc-gKXOVf bBKeoX"
    >
      <div
        className="sc-eCYdqJ sc-papXJ kJXsff dSWLEt"
      >
        <span
          className="sc-bczRLJ YxcWK"
        >
          BTC
        </span>
        <svg
          height="1.5rem"
          viewBox="0 0 24 24"
          width="1.5rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs>
            <style>
              .up { fill: #ae5467;} .down { fill: #29a5e9;}
            </style>
          </defs>
          <g
            id="a"
          />
          <g
            id="b"
          >
            <polygon
              className="up"
              points="8 14 16 14 12 19 8 14"
            />
            <polygon
              className="down"
              points="16 10 8 10 12 5 16 10"
            />
          </g>
        </svg>
        <span
          className="sc-bczRLJ YxcWK"
        >
          USD
        </span>
      </div>
      <div>
        <span
          className="sc-bczRLJ bYxRpA"
        >
          No Data
        </span>
      </div>
    </div>
  </div>,
  <div
    className="sc-jSMfEi eUtByO"
    color="#1f2422"
  >
    <div
      className="sc-gKXOVf bBKeoX"
    >
      <div
        className="sc-eCYdqJ sc-papXJ kJXsff dSWLEt"
      >
        <span
          className="sc-bczRLJ YxcWK"
        >
          BTC
        </span>
        <svg
          height="1.5rem"
          viewBox="0 0 24 24"
          width="1.5rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs>
            <style>
              .up { fill: #ae5467;} .down { fill: #29a5e9;}
            </style>
          </defs>
          <g
            id="a"
          />
          <g
            id="b"
          >
            <polygon
              className="up"
              points="8 14 16 14 12 19 8 14"
            />
            <polygon
              className="down"
              points="16 10 8 10 12 5 16 10"
            />
          </g>
        </svg>
        <span
          className="sc-bczRLJ YxcWK"
        >
          USD
        </span>
      </div>
      <div>
        <span
          className="sc-bczRLJ bYxRpA"
        >
          No Data
        </span>
      </div>
    </div>
  </div>,
  <div
    className="sc-jSMfEi eUtByO"
    color="#1f2422"
  >
    <div
      className="sc-gKXOVf bBKeoX"
    >
      <div
        className="sc-eCYdqJ sc-papXJ kJXsff dSWLEt"
      >
        <span
          className="sc-bczRLJ YxcWK"
        >
          BTC
        </span>
        <svg
          height="1.5rem"
          viewBox="0 0 24 24"
          width="1.5rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs>
            <style>
              .up { fill: #ae5467;} .down { fill: #29a5e9;}
            </style>
          </defs>
          <g
            id="a"
          />
          <g
            id="b"
          >
            <polygon
              className="up"
              points="8 14 16 14 12 19 8 14"
            />
            <polygon
              className="down"
              points="16 10 8 10 12 5 16 10"
            />
          </g>
        </svg>
        <span
          className="sc-bczRLJ YxcWK"
        >
          USD
        </span>
      </div>
      <div>
        <span
          className="sc-bczRLJ bYxRpA"
        >
          No Data
        </span>
      </div>
    </div>
  </div>,
]
`;
